soci/index_builder.go

```diff
@@ -0,0 +1,161 @@
+package soci
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/containerd/containerd/content"
+	"github.com/containerd/containerd/images"
+	"github.com/containerd/containerd/platforms"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+	"golang.org/x/sync/errgroup"
+)
+
+type IndexBuilder struct {
+	contentStore        content.Store
```

Comment:
Do we want to use `content.Store` here? As @Kern-- mentioned, `content.Store` might be too large of an interface to use here. We can instead use ORAS's [content.Storage](https://pkg.go.dev/oras.land/oras-go/v2@v2.0.0-rc.4/content#Storage) or define our own.

Reply:
Agree that it's a big interface. Would you mind if I do a fast follow on that? The reasons are:
1. I don't know how yet
2. I don't want to introduce too much changes in a single PR.

Reply:
Sure.